docs/orchestration.asciidoc

```diff
@@ -0,0 +1,104 @@
+[id="{p}-orchestration"]
+=== NodeSets orchestration
+
+[id="{p}-nodesets"]
+==== NodeSets overview
+
+The Elasticsearch cluster is specified using a list of NodeSets. Each NodeSet represents a group of Elasticsearch nodes sharing the same specification (both Elasticsearch configuration and Kubernetes Pod configuration).
+
+[source,yaml,subs="attributes"]
+----
+apiVersion: elasticsearch.k8s.elastic.co/{eck_crd_version}
+kind: Elasticsearch
+metadata:
+  name: quickstart
+spec:
+  version: {version}
+  nodeSets:
+  - name: master-nodes
+    count: 3
+    config:
+      master: true
+      data: false
+    volumeClaimTemplates:
+    - metadata:
+        name: elasticsearch-data
+      spec:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: 10Gi
+        storageClassName: standard
+  - name: data-nodes
+    count: 10
+    config:
+      master: false
+      data: true
+    volumeClaimTemplates:
+    - metadata:
+        name: elasticsearch-data
+      spec:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: 1000Gi
+        storageClassName: standard
+----
+
+The Elasticcsearch resource above defines two NodeSets: one for master nodes, using 10Gi volumes, and one for data nodes, using 1000Gi volumes. The Elasticsearch cluster will be composed of 13 nodes: 3 master nodes and 10 data nodes.
+
+[id="{p}-upgrading"]
+==== Upgrading the cluster
+
+ECK handles smooth upgrades from one cluster specification to another.
+
+Some examples based on the Elasticsearch specification above:
+
+- To add 5 additional data nodes, simply change `count: 10` to `count: 15` in the `data-nodes` NodeSet, then apply the Elasticsearch resource.
+- To increase the RAM memory limit of data Nodes to 32Gi, link:k8s-managing-compute-resources.html[set a different resources limits] in the existing `data-nodes` NodeSet PodTemplate.
+- To replace dedicated master and dedicated data nodes by nodes having both master and data roles, replace the 2 existing NodeSets by a single one with a different name and the corresponding Elasticsearch configuration settings.
+
+No downtime is expected when NodeSet changes are orchestrated by ECK. It ensures that:
+
+- data from a node to remove is completely migrated away to other nodes before the node is actually removed
+- Elasticsearch orchestration settings such as `discovery.seed_hosts`, `cluster.initial_master_nodes`, `discovery.zen.minimum_master_nodes`, `_cluster/voting_config_exclusions` correctly account for any change in the cluster
+- rolling upgrades are performed in a safe way, reusing PersistentVolumes of the upgraded Elasticsearch nodes
```

Comment:
```suggestion
- rolling upgrades are performed safely, reusing PersistentVolumes of the upgraded Elasticsearch nodes (i.e. in-place upgrades rather than grow-and-shrink)
```

I also wonder if it is worth breaking these up into two separate bullets, as we could also do rolling upgrades unsafely while still reusing PVs

Reply:
"in-place upgrades rather than grow-and-shrink" is our own terminology that we would then need to explain. Not sure that's necessary?